pkg/util/util_test.go

```diff
@@ -26,17 +26,86 @@ const validCluster = `
 apiVersion: "cluster.k8s.io/v1alpha1"
 kind: Cluster
 metadata:
-  name: cluster1 
+  name: cluster1
 spec:`
 
-const validMachines = `
+const validMachines1 = `
 items:
 - apiVersion: "cluster.k8s.io/v1alpha1"
   kind: Machine
   metadata:
     name: machine1
   spec:`
 
+const validMachines2 = `
+---
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: Machine
+metadata:
+  name: machine1
+---
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: Machine
+metadata:
+  name: machine2`
+
+const validUnified1 = `
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: Cluster
+metadata:
+  name: cluster1
+---
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineList
+items:
+- apiVersion: "cluster.k8s.io/v1alpha1"
+  kind: Machine
```

Comment:
Just an idea: given that this is a MachineList, are `apiVersion` and `kind` necessary above or could be assumed from above?

Reply:
Works, but 
```
apiVersion: "cluster.k8s.io/v1alpha1"
kind: MachineList
```
is required in that use case

Reply:
It works either way, test added.